rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/custom_participant_info.hpp

```diff
@@ -235,16 +273,115 @@ class ParticipantListener : public eprosima::fastdds::dds::DomainParticipantList
           is_reader);
       } else {
         context->graph_cache.remove_entity(
-          rmw_fastrtps_shared_cpp::create_rmw_gid(
-            identifier_,
-            proxyData.guid()),
-          is_reader);
+            rmw_fastrtps_shared_cpp::create_rmw_gid(identifier_,
+                                                    proxyData.guid()),
+            is_reader);
+      }
+    }
+  }
+
+  std::unordered_set<std::string>
+  get_other_static_peers(
+    const std::map<std::string, std::vector<uint8_t>> &map) {
+    std::unordered_set<std::string> result;
+
+    auto static_peers_entry = map.find("staticpeers");
+    if (static_peers_entry != map.end()) {
+      auto static_peers_string = std::string(static_peers_entry->second.begin(),
+                                             static_peers_entry->second.end());
+      std::string delimiter(",");
+      size_t start = 0;
+      size_t end = 0;
+      while ((end = static_peers_string.find(delimiter, start)) !=
+             std::string::npos) {
```

Comment:
nitpick:

```suggestion
      while (
        (end = static_peers_string.find(delimiter, start)) != std::string::npos
      ) {
```

Reply:
Also, you might consider just using https://github.com/ros2/rcpputils/blob/39b20134e571ba74baa7c77750eab586da90b7a5/include/rcpputils/split.hpp#L79-L95, because I think the crux of this block of code is splitting the string right?

Reply:
I considered that, but this is more efficient as it does not do any memory allocations so I kept it.

Reply:
Whitespace change in dfd7dd2